providers/structured/provider.go

```diff
@@ -47,16 +50,25 @@ func (p StructuredOutputLogProvider) GetRecordEventCalls() []RecordEventCallArgs
 
 // NewStructuredOutputLogProvider returns a LogProvider which records all calls made to it.
 func NewStructuredOutputLogProvider() *StructuredOutputLogProvider {
+	return LogProvider(nil).(*StructuredOutputLogProvider)
+}
+
+func LogProvider(nextProvider providers.LogProvider) providers.LogProvider {
 	rawLogCalls := map[providers.RawLogCallType][]RawLogCallArgs{}
 	for _, callType := range providers.RawLogCallTypes() {
 		rawLogCalls[callType] = []RawLogCallArgs{}
 	}
 
-	return &StructuredOutputLogProvider{
+	lp := &StructuredOutputLogProvider{
 		rawLogCalls:      rawLogCalls,
 		recordCalls:      []RecordCallArgs{},
 		recordEventCalls: []RecordEventCallArgs{},
 	}
+
+	if nextProvider != nil {
```

Comment:
Possible to test the behavior of this block in the unit tests? The rest of this method is already covered, so you'd just need to add a test for `nextProvider == nil`/another for `nextProvider != nil`.